### ISSUE TITLE
no-extra-parens does not allow logical expressions to be wrapped to avoid no-mixed-operators

### ISSUE THREAD

**Issue by dackmin** [eslint/eslint#13413]
**Tell us about your environment**

* **ESLint Version:** 7.2.0
* **Node Version:** 12.14.1
* **npm Version:** 6.13.4
* **yarn Version:** 1.22.4

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  parserOptions: {
    ecmaVersion: 2020,
    sourceType: 'module',
  },
  rules: {
    'no-extra-parens': [1, 'all'],
    'no-mixed-operators': 2,
  },
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
export const applyPlatformFilter = platform =>
  d => platform === 'total' ||
    (platform === 'mobile' && d.mobile === true) ||
    (platform === 'desktop' && d.mobile === false);
```

```bash
./node_modules/.bin/eslint .
```

Also tested on your [demo](https://eslint.org/demo#eyJ0ZXh0IjoiZXhwb3J0IGNvbnN0IGFwcGx5UGxhdGZvcm1GaWx0ZXIgPSBwbGF0Zm9ybSA9PlxuICBkID0+IHBsYXRmb3JtID09PSAndG90YWwnIHx8XG4gICAgKHBsYXRmb3JtID09PSAnbW9iaWxlJyAmJiBkLm1vYmlsZSA9PT0gdHJ1ZSkgfHxcbiAgICAocGxhdGZvcm0gPT09ICdkZXNrdG9wJyAmJiBkLm1vYmlsZSA9PT0gZmFsc2UpOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTEsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsiY29uc3RydWN0b3Itc3VwZXIiOjIsImZvci1kaXJlY3Rpb24iOjIsImdldHRlci1yZXR1cm4iOjIsIm5vLWFzeW5jLXByb21pc2UtZXhlY3V0b3IiOjIsIm5vLWNhc2UtZGVjbGFyYXRpb25zIjoyLCJuby1jbGFzcy1hc3NpZ24iOjIsIm5vLWNvbXBhcmUtbmVnLXplcm8iOjIsIm5vLWNvbmQtYXNzaWduIjoyLCJuby1jb25zdC1hc3NpZ24iOjIsIm5vLWNvbnN0YW50LWNvbmRpdGlvbiI6Miwibm8tY29udHJvbC1yZWdleCI6Miwibm8tZGVidWdnZXIiOjIsIm5vLWRlbGV0ZS12YXIiOjIsIm5vLWR1cGUtYXJncyI6Miwibm8tZHVwZS1jbGFzcy1tZW1iZXJzIjoyLCJuby1kdXBlLWVsc2UtaWYiOjIsIm5vLWR1cGUta2V5cyI6Miwibm8tZHVwbGljYXRlLWNhc2UiOjIsIm5vLWVtcHR5IjoyLCJuby1lbXB0eS1jaGFyYWN0ZXItY2xhc3MiOjIsIm5vLWVtcHR5LXBhdHRlcm4iOjIsIm5vLWV4LWFzc2lnbiI6Miwibm8tZXh0cmEtYm9vbGVhbi1jYXN0IjoyLCJuby1leHRyYS1zZW1pIjoyLCJuby1mYWxsdGhyb3VnaCI6Miwibm8tZnVuYy1hc3NpZ24iOjIsIm5vLWdsb2JhbC1hc3NpZ24iOjIsIm5vLWltcG9ydC1hc3NpZ24iOjIsIm5vLWlubmVyLWRlY2xhcmF0aW9ucyI6Miwibm8taW52YWxpZC1yZWdleHAiOjIsIm5vLWlycmVndWxhci13aGl0ZXNwYWNlIjoyLCJuby1taXNsZWFkaW5nLWNoYXJhY3Rlci1jbGFzcyI6Miwibm8tbWl4ZWQtc3BhY2VzLWFuZC10YWJzIjoyLCJuby1uZXctc3ltYm9sIjoyLCJuby1vYmotY2FsbHMiOjIsIm5vLW9jdGFsIjoyLCJuby1wcm90b3R5cGUtYnVpbHRpbnMiOjIsIm5vLXJlZGVjbGFyZSI6Miwibm8tcmVnZXgtc3BhY2VzIjoyLCJuby1zZWxmLWFzc2lnbiI6Miwibm8tc2V0dGVyLXJldHVybiI6Miwibm8tc2hhZG93LXJlc3RyaWN0ZWQtbmFtZXMiOjIsIm5vLXNwYXJzZS1hcnJheXMiOjIsIm5vLXRoaXMtYmVmb3JlLXN1cGVyIjoyLCJuby11bmRlZiI6Miwibm8tdW5leHBlY3RlZC1tdWx0aWxpbmUiOjIsIm5vLXVucmVhY2hhYmxlIjoyLCJuby11bnNhZmUtZmluYWxseSI6Miwibm8tdW5zYWZlLW5lZ2F0aW9uIjoyLCJuby11bnVzZWQtbGFiZWxzIjoyLCJuby11bnVzZWQtdmFycyI6Miwibm8tdXNlbGVzcy1jYXRjaCI6Miwibm8tdXNlbGVzcy1lc2NhcGUiOjIsIm5vLXdpdGgiOjIsInJlcXVpcmUteWllbGQiOjIsInVzZS1pc25hbiI6MiwidmFsaWQtdHlwZW9mIjoyLCJuby1leHRyYS1wYXJlbnMiOjIsIm5vLW1peGVkLW9wZXJhdG9ycyI6Mn0sImVudiI6e319fQ==).

**What did you expect to happen?**
Parentheses around logical expressions to avoid mixed operators should have been valid.

**What actually happened? Please include the actual, raw output from ESLint.**
These mandatory parentheses are detected as _extra_:

```
/[...]/data.js
  3:5  warning  Unnecessary parentheses around expression  no-extra-parens
  4:5  warning  Unnecessary parentheses around expression  no-extra-parens
```
When using the `--fix` argument, the _extra_ parentheses are removed but then:

```
/[...]/data.js
  2:29  error  Unexpected mix of '||' and '&&'  no-mixed-operators
  3:27  error  Unexpected mix of '||' and '&&'  no-mixed-operators
  3:48  error  Unexpected mix of '||' and '&&'  no-mixed-operators
  4:28  error  Unexpected mix of '||' and '&&'  no-mixed-operators
```

**Are you willing to submit a pull request to fix this bug?**

I'm not sure if I can be of any help, but I can try 🤷‍♂️

**Comment by dackmin** [comment]
my bad, I completely forgot about `nestedBinaryExpressions`
